# DP: Fix PR ld/14918: linking with -flto always links in libgcc_s.so

2012-12-06  H.J. Lu  <hongjiu.lu@intel.com>

	PR ld/14918
	* plugin.c (plugin_opt_plugin_arg): Ignore -pass-through=.

diff --git a/ld/plugin.c b/ld/plugin.c
index da99e77..2966f0f 100644
--- a/ld/plugin.c
+++ b/ld/plugin.c
@@ -217,6 +220,17 @@ plugin_opt_plugin_arg (const char *arg)
   if (!last_plugin)
     return set_plugin_error (_("<no plugin>"));
 
+  /* Ignore -pass-through= from GCC driver.  */
+  if (*arg == '-')
+    {
+      const char *p;
+      for (p = arg + 1; p; p++)
+	if (*p != '-')
+	  break;
+      if (strncmp (p, "pass-through=", 13) == 0)
+	return 0;
+    }
+
   newarg = xmalloc (sizeof *newarg);
   newarg->arg = arg;
   newarg->next = NULL;
